keyboards/cutie_club/keebcats/gina/gina.c

```diff
@@ -0,0 +1,116 @@
+/* Copyright 2023 Cutie Club
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "gina.h"
+
+#include "indicator_leds.h"
+
+#include "via.h"
+#include "i2c_master.h"
+#include "quantum/rgblight/rgblight.h"
+
+static keyboard_config_t keyboard_config;
+
+void keyboard_pre_init_kb(void) {
+    i2c_init();
+
+    keyboard_config.raw = eeconfig_read_kb();
+
+    enum indicator_key indicator_key = keyboard_config.indicator_key;
+
+    if (indicator_key != NUM_LOCK && indicator_key != SCROLL_LOCK && indicator_key != CAPS_LOCK) {
+        indicator_key                 = DEFAULT_INDICATOR_LED;
+        keyboard_config.indicator_key = indicator_key;
+        eeconfig_update_kb(keyboard_config.raw);
+    }
+    set_enabled_indicator_key(keyboard_config.indicator_key);
+}
+
+#ifndef VIA_ENABLED
```

Comment:
Implementations of RGB indicators for VIA are no longer being accepted to the repo.

Reply:
Thank you for the heads-up! My apologies, I must have missed this announcement (´• ω •`).

Is there anywhere I can read more about the reasoning behind this?

Part of the reason for the custom RGB handling in this case is allowing users to change the colour of an LED without having to have the LED switched on through the `rgblight_sethsv_noeeprom` function, would this be something I can raise a PR to "fix" or is it intended behaviour?

Reply:
It's one of the PRs, but basically, all of the code is basically just copy-pasta.  So, it should be moved to core/via.c, rather than implemented on each board.   However, that has not been done.

For rgblight, you can set the clipping range, as well as use the rgblight lighting indicators stuff, as well, in theory.